mgmt-lambda/handlers/updateHandler.ts

```diff
@@ -127,65 +135,26 @@ async function updateLambdaFunctionCode(lambdaClient: LambdaClient, functionName
   })
   console.info('Sending update command to Lambda runtime')
   const result = await lambdaClient.send(command)
-  console.info(`Got Lambda function update result, functionARN: ${result.FunctionArn}`)
+
+  if (!result) {
+    throw new ApiException(ErrorCode.FunctionARNNotFound)
+  }
 
   if (!result.FunctionArn) {
-    throw new Error('Function ARN not found after update')
+    throw new ApiException(ErrorCode.FunctionARNNotFound)
   }
 
+  console.info(`Got Lambda function update result, functionARN: ${result.FunctionArn}`)
+
   return result.FunctionArn
 }
 
-async function checkLambdaFunctionExistence(client: LambdaClient, functionName: string): Promise<boolean> {
-  const params: GetFunctionCommandInput = {
-    FunctionName: functionName,
-  }
-  const command = new GetFunctionCommand(params)
+/**
+ * @throws {import('@aws-sdk/client-lambda').LambdaServiceException}
+ */
+async function checkIfLambdaFunctionWithNameExists(client: LambdaClient, functionName: string): Promise<boolean> {
+  const command = new GetFunctionCommand({ FunctionName: functionName })
   const result: GetFunctionCommandOutput = await client.send(command)
-  if (!result.Configuration?.FunctionArn) {
-    return false
-  }
-  return true
-}
 
-async function handleException(status: string, message: string): Promise<APIGatewayProxyResult> {
-  const body = {
-    status: status,
-    error: message,
-  }
-  return {
-    statusCode: 500,
-    body: JSON.stringify(body),
-    headers: {
-      'content-type': 'application/json',
-    },
-  }
-}
-
-async function handleFailure(message?: any): Promise<APIGatewayProxyResult> {
-  const body = {
-    status: 'Update failed',
-    error: message,
-  }
-  return {
-    statusCode: 500,
-    body: JSON.stringify(body),
-    headers: {
-      'content-type': 'application/json',
-    },
-  }
-}
-
-async function handleSuccess(): Promise<APIGatewayProxyResult> {
-  const body = {
-    status: 'Update completed',
-    error: null,
-  }
-  return {
-    statusCode: 200,
-    body: JSON.stringify(body),
-    headers: {
-      'content-type': 'application/json',
-    },
-  }
+  return typeof result?.Configuration?.FunctionArn !== 'undefined'
```

Comment:
Why only check for `undefined` and not also check for `null` here? Something like `return result?.Configuration?.FunctionArn != null`